### PR TITLE
Adding database flags for pootle init

### LINKED ISSUE #4015: Add --host --database --username and --password flags to pootle init
To make it easier when setting up mysql/postgresql for the first time

### PR DESCRIPTION
**Pull request by phlax**
This commit adds the following flags to pootle init for setting up
database config from the command line:
 - --db-name
 - --db-user
 - --db-password
 - --db-host
 - --db-port

It also does the following:
 - renames settings/90-local.conf.sample to
   settings/90-local.conf.template to reflect that its template
 - Adds documentation for the `pootle init` command

Fixes #4015

### CODE REVIEW

File: docs/server/installation.rst [hunk 1]
```diff
@@ -212,6 +212,10 @@ an alternative path as an argument if required.
 .. warning:: This default configuration is enough to experiment with Pootle.
    **Don't use this configuration in a production environment**.
 
+You can specify the parameters to set up your database if you don't want to use
```
Comment (julen) [inline docs/server/installation.rst]: Hmmm reading this I have the feeling that a SQLite setup is a blessed option. Can we maybe reword it to omit SQLite at all?

Comment (phlax) [inline docs/server/installation.rst]: how do you mean @julen - i dont quite follow

File: pootle/runner.py [hunk 2]
```diff
@@ -58,13 +70,19 @@ def init_settings(settings_filepath, template_filename, db="sqlite"):
...
-                                "'NAME': ''")
+    context = {
+        "default_key": ("'%s'"
+                        % b64encode(os.urandom(KEY_LENGTH)).decode("utf-8")),
+        "db_engine": "'transaction_hooks.backends.%s'" % db_module,
+        "db_name": db_name,
+        "db_user": db_user,
+        "db_password": db_password,
+        "db_host": db_host,
+        "db_port": db_port
```
Comment (julen) [inline pootle/runner.py]: Always use a trailing comma, it reduces the size of future diffs.

Comment (phlax) [inline pootle/runner.py]: ill add

File: docs/server/installation.rst (hunk 1 above)
Comment (julen) [inline docs/server/installation.rst]: *default sqlite setup* sounds to me like an option we'd recommend.

Comment (phlax) [inline docs/server/installation.rst]: k - got it - ill change to just "default setup"

File: docs/server/commands.rst [hunk 3]
```diff
@@ -500,6 +500,51 @@ Manually Installing Pootle
...
+Create the initial configuration for Pootle.
+
+Available options:
+
+:option:`--config`
+  The configuration file to write to.
+
+  Default: ``~/.pootle/pootle.conf``.
+
+.. versionchanged:: 2.7.1
```
Comment (unho) [inline docs/server/commands.rst]: What has changed?

File: docs/server/installation.rst (hunk 1 above)
Comment (unho) [inline docs/server/installation.rst]: &lt;spam>**Just lets drop SQLite!!!**&lt;/spam>

File: docs/server/commands.rst (hunk 3 above)
Comment (phlax) [inline docs/server/commands.rst]: Additional options

Comment (phlax) [inline docs/server/commands.rst]: maybe i should add versionchanged on each option?

File: pootle/runner.py [hunk 4]
```diff
@@ -58,13 +70,19 @@ def init_settings(settings_filepath, template_filename, db="sqlite"):
...
+    }
 
+    fp = open(settings_filepath, 'w')
```
Comment (unho) [inline pootle/runner.py]: Since we are rewriting perhaps we can use `with`.

Comment (phlax) [inline pootle/runner.py]: agreed

File: pootle/runner.py [hunk 5]
```diff
@@ -148,16 +185,14 @@ def run_app(project, default_settings_path, settings_template,
...
+            init_settings(config_path, settings_template,
+                          args.db, args.db_name, args.db_user,
+                          args.db_password, args.db_host, args.db_port)
         except (IOError, OSError) as e:
             raise e.__class__('Unable to write default settings file to %r'
                 % config_path)
 
-        if args.db in ["mysql", "postgresql"]:
-            print("Configuration file created at %r: you must now update "
-                  "the settings for %s database" % (config_path, args.db))
```
Comment (unho) [inline pootle/runner.py]: If we specify `mysql` but we don't provide a `db_pasword` we should warn about settings not being complete, isn't it?

Comment (phlax) [inline pootle/runner.py]: i did consider this but that would actually be a valid configuration.
Ill add a warning along the lines of - "you have not specified a password for your database. you may want to update the settings now"

File: docs/server/commands.rst (hunk 3 above)
Comment (unho) [inline docs/server/commands.rst]: What about `versionadded` on each new option, or a versionadded block (http://sphinx-doc.org/markup/para.html#directive-versionadded) listing the new options?

File: pootle/runner.py (hunk 5 above)
Comment (unho) [inline pootle/runner.py]: Good.

File: docs/server/commands.rst (hunk 3 above)
Comment (phlax) [inline docs/server/commands.rst]: cool - ta - ill update with version*added*

**Comment by unho** [comment]
lgtm

**Comment by dwaynebailey** [comment]
lgtm
